service/geopmdpy/system_files.py

```diff
@@ -161,25 +162,26 @@ def secure_make_file(path, contents):
 
     The file is created by first writing the output to a temporary
     file with restricted permissions (mode 0o600) within the same
-    directory, and then renaming the file after it is closed.  This
-    temporary file with have the path `<path>-<uuid>-tmp`.
+    directory, and then renaming the file after it is closed.
 
     The rename operation is atomic, and will overwrite any existing
     file located in the specified path.  The permissions on the output
     file are mode 0o600 with uid and gid matching the process values.
 
+    The temporary file has the prefix `tmp.`. This file may persist if
+    the program exits abruptly.
+
     Args:
         path (str): The path where the file is created
 
         contents (str): The contents of the created file
 
     """
-    temp_path = f'{path}-{uuid.uuid4()}-tmp'
-    old_mask = os.umask(0o077)
+    old_mask = os.umask(0o477)
     try:
-        with open(os.open(temp_path, os.O_CREAT | os.O_WRONLY, 0o600), 'w') as file:
+        with tempfile.NamedTemporaryFile(mode="w+", prefix="tmp.", dir=os.path.dirname(path), delete=False) as file:
```

Comment:
I think the reason for using os.open() directly is to enable the use of the flags here.  Does NamedTemporaryFile use your umask, or does it always create with 0o600 permissions?